### PR TITLE
Cumulus 1556 add terraform release scripts

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-1556](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1556)

## Changes

- Adds bamboo script that creates a release, packages the terraform modules/lambdas and uploads the release artifact.

### CODE REVIEW

File: bamboo/deploy-to-npm.sh [hunk 1]
```diff
@@ -17,3 +17,4 @@ npx lerna publish \
   --dist-tag=${NPM_TAG} \
   --exact
 
+. ./bamboo/create-release.sh
```
Comment (Jkovarik) [inline bamboo/deploy-to-npm.sh]: Decided to explicitly call this script post deploy-to-npm as: 

Updating the release process with another phase is workable, however for previous branches to pass we'd have to || true pass it (or detect existence and pass). 

I think the correct change is to rename both scripts to something like `publish_release_artifacts`, however I'd like to attempt that when it's less likely to impact objective work.

File: bamboo/create-release.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
```
Comment (Jkovarik) [inline bamboo/create-release.sh]: For a sample of what this script *does*, you can take a look at the release I generated in testing on my fork: 

https://github.com/Jkovarik/cumulus/releases

File: bamboo/create-release.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export VERSION=$(jq --raw-output .version lerna.json)
+
+## Create Release
+   export RELEASE_URL=$(curl -H\
+  "Authorization: token $TOKEN"\
+   -d "{\"tag_name\": \"v$VERSION\", \"target_commitsh\": \"v$VERSION\", \"name\": \"v$VERSION\", \"body\": \"Release v$VERSION\" }"\
+   -H "Content-Type: application/json"\
+   -X POST\
+   https://api.github.com/repos/jkovarik/cumulus/releases |grep \"url\" |grep releases |sed -e 's/.*\(https.*\)\"\,/\1/'| sed -e 's/api/uploads/')
```
Comment (markdboyd) [inline bamboo/create-release.sh]: Need to change `jkovarik` to `nasa`

File: docs/development/release.md [hunk 4]
```diff
@@ -98,6 +98,10 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
 
 ### 9. Create a new Cumulus release on github
 
+The release scripts will automatically create a release based on the release version tag, as well as upload a multi-module Terraform .zip artifact containing filtered copies  `tf-modules`, `packages` and `tasks` directories for use as as Terraform module sources.
```
Comment (markdboyd) [inline docs/development/release.md]:
```suggestion
The release scripts will automatically create a release based on the release version tag, as well as upload a multi-module Terraform `.zip` artifact containing filtered copies of the `tf-modules`, `packages` and `tasks` directories for use as as Terraform module sources.
```

File: docs/development/release.md [hunk 5]
```diff
@@ -98,6 +98,10 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
 
 ### 9. Create a new Cumulus release on github
 
+The release scripts will automatically create a release based on the release version tag, as well as upload a multi-module Terraform `.zip` artifact containing filtered copies of the `tf-modules`, `packages` and `tasks` directories for use as as Terraform module sources.
+
+Additionally, the following resources should be manually created and added to the distribution:
```
Comment (markdboyd) [inline docs/development/release.md]:
```suggestion
Additional resources should be manually created and added to the release following the steps below.
```

File: docs/development/release.md [hunk 6]
```diff
@@ -98,21 +98,7 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
...
-This will generate a file called
-`tf-modules/distribution/dist/terraform-aws-cumulus-distribution.zip`, which
-should be uploaded as an asset of the Github release.
-
-Similarily, the S3 Replicator Terraform module can be created by running:
-
-```bash
-$ (cd tf-modules/s3-replicator && ./bin/build-tf-module.sh)
-```
+The release scripts will automatically create a release based on the release version tag, as well as upload a multi-module Terraform `.zip` artifact containing filtered copies of the `tf-modules`, `packages` and `tasks` directories for use as as Terraform module sources.     Just make sure to verify the appropriate .zip files are present on Github.
```
Comment (markdboyd) [inline docs/development/release.md]:
```suggestion
The CI release scripts will automatically create a release based on the release version tag, as well as uploading release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:

- A multi-module Terraform `.zip` artifact containing filtered copies of the `tf-modules`, `packages`, and `tasks` directories for use as Terraform module sources. 
- A S3 replicator module
- A workflow module
- A distribution API module
- An ECS service module

Just make sure to verify the appropriate .zip files are present on Github after the release process is complete.
```

Oh yes, the king of nitpicks is in

Comment (Jkovarik) [reply docs/development/release.md]: Hmm, this doesn't *have* to run in CI though.

File: docs/development/release.md [hunk 7]
```diff
@@ -98,21 +98,15 @@ Bamboo will build and run lint, audit and unit tests against that tagged release
 
 ### 9. Create a new Cumulus release on github
 
-The Cumulus Distribution Terraform module can be created by running:
+The CI release scripts will automatically create a release based on the release version tag, as well as uploading release artifacts to the Github release for the Terraform modules provided by Cumulus. The Terraform release artifacts include:
```
Comment (markdboyd) [inline docs/development/release.md]: Yeah, if it's not accurate to say CI then that should be removed

**Pull request merged**